digdag-core/src/main/java/io/digdag/core/database/DatabaseSessionStoreManager.java

```diff
@@ -288,6 +287,38 @@ public boolean isAnyNotDoneAttempts()
     }
 
     @Override
+    public List<StoredSessionAttempt> findAttemptsCreatedBeforeWithState(Instant createdBefore, int state, long lastId, int limit)
```

Comment:
`CREATE INDEX CONCURRENTLY` is a blocking operation although it doesn't lock a table. So, yes, digdag still waits. Moreover, `CREATE INDEX CONCURRENTLY` can't run in a transaction.

I agree that separation of migration is eventually necessary. I hope that it's not now.